### PR TITLE
feat: Add 1.22 and 1.25 rules according to Deprecated API Migration guide

### LINKED ISSUE #135: Review and add rules for Deprecated API Migration Guide
K8s docs now have handy page [Deprecated API Migration Guide](https://kubernetes.io/docs/reference/using-api/deprecation-guide/).

Go over it and make sure we support those resources or add them otherwise.

### PR DESCRIPTION
**Pull request by zetaab**
fixes #135

### CODE REVIEW

File: pkg/rules/rego/deprecated-1-25.rego [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			"since": "1.19",
+		},
+		"PodDisruptionBudget": {
+			"old": ["policy/v1beta1"],
+			"new": "policy/v1",
+			"since": "1.21",
+		},
+		"PodSecurityPolicy": {
+			"old": ["policy/v1beta1"],
+			"new": "",
```
Comment (zetaab) [inline pkg/rules/rego/deprecated-1-25.rego:46]: how to mark api that does not have replacement?

**Comment by david-doit-intl** [comment]
@zetaab thank you for this we will review it!

File: pkg/rules/rego/deprecated-1-22.rego [hunk 2]
```diff
@@ -22,10 +22,15 @@ deprecated_resource(r) = api {
 deprecated_api(kind, api_version) = api {
 	deprecated_apis = {
 		"Ingress": {
-			"old": ["extensions/v1beta1"],
-			"new": "networking.k8s.io/v1beta1",
+			"old": ["extensions/v1beta1", "networking.k8s.io/v1beta1"],
```
[the remark this answers is not in the document]
Comment (stepanstipl) [reply pkg/rules/rego/deprecated-1-22.rego:25]: Yep, currently, we don't support this scenario where there could be multiple versions to replace with, so I think this looks ok.

File: pkg/rules/rego/deprecated-1-25.rego (hunk 1 above)
Comment (stepanstipl) [reply pkg/rules/rego/deprecated-1-25.rego:46]: Same, currently there's no direct way to support this, this should be good for now (need to test).

**Comment by stepanstipl** [comment]
Thanks @zetaab, this is really great. 👍  I think it's pretty much good to go. I would like to cover these with tests, but we don't have quite a good pattern for that yet, so let's leave that part for another PR.

What is currently failing is a commit message - we have a [naming convention](https://github.com/doitintl/kube-no-trouble#commit-messages) for that. I would suggest splitting this into 2 commits and naming smth like:
```
feat: Add 1.22 rules according to Deprecated API Migration guide
feat: Add 1.25 rules according to Deprecated API Migration guide
```

**Comment by david-doit-intl** [comment]
thank you @zetaab running tests now!

File: pkg/rules/rego/deprecated-1-22.rego [hunk 3]
```diff
@@ -41,11 +46,36 @@ deprecated_api(kind, api_version) = api {
...
+			"old": ["apiregistration.k8s.io/v1beta1"],
+			"new": "apiregistration.k8s.io/v1",
+			"since": "1.10",
+		},
+		"CustomResourceDefinition": {
+			"old": ["apiextensions.k8s.io/v1beta1"],
+			"new": "apiextensions.k8s.io/v1",
+			"since": "1.16",
+		},
+		"Webhooks": {
```
Comment (stepanstipl) [inline pkg/rules/rego/deprecated-1-22.rego:74]: This actually needs to be split in 2 `MutatingWebhookConfiguration` and `ValidatingWebhookConfiguration` resources.

**Review by david-doit-intl — DISMISSED** [review]

File: pkg/rules/rego/deprecated-1-22.rego [hunk 4]
```diff
@@ -41,11 +46,36 @@ deprecated_api(kind, api_version) = api {
...
 		},
-		"LeaseList": {
-			"old": ["coordination.k8s.io/v1beta1"],
-			"new": "coordination.k8s.io/v1",
+		"PriorityClass": {
+			"old": ["scheduling.k8s.io/v1beta1"],
+			"new": "scheduling.k8s.io/v1",
+			"since": "1.14",
+		},
+		"RBAC": {
```
Comment (stepanstipl) [inline pkg/rules/rego/deprecated-1-22.rego:54]: This needs to be expanded into 4 actual resources: 
- `ClusterRole`
- `ClusterRoleBinding`
- `Role`
- `RoleBinding`

**Review by stepanstipl — COMMENTED** [review]
Looking at this more in details, some further comments inline plust:

- We're missing `LocalSubjectAccessReview` and `SelfSubjectAccessReview`
- Storage resources:
> The storage.k8s.io/v1beta1 API version of CSIDriver, CSINode, StorageClass, and 
>  VolumeAttachment will no longer be served in v1.22.
>
> Migrate manifests and API clients to use the storage.k8s.io/v1 API version
> CSIDriver is available in storage.k8s.io/v1 since v1.19.
> CSINode is available in storage.k8s.io/v1 since v1.17
> StorageClass is available in storage.k8s.io/v1 since v1.6
> VolumeAttachment is available in storage.k8s.io/v1 v1.13

**Comment by stepanstipl** [comment]
This all has now been covered by a separate PRs: #163, #165, #166, #167, #168, #169, #170, #171, #172, #173, #174 and #175.

If you find anything missing/not working, please raise a separate PR/issue. Thanks everyone a lot for help with this!

Related to #135

**Pull request closed without merging**